messaging/api/src/main/java/org/eclipse/microprofile/reactive/messaging/Event.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.eclipse.microprofile.reactive.messaging;
+
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An abstract event envelope, modeled after the
+ * <a href="https://github.com/cloudevents/spec/blob/master/spec.md">CNCF CloudEvent spec</a>.
+ *
+ */
+public interface Event<T> extends Message<T> {
+    /**
+     * Type of occurrence which has happened. Often this property is used for routing, observability, policy enforcement, etc.
```

Comment:
Perhaps doesn't need to be mentioned yet, but I think an important use of this property will be in deserialization - ie, deserializing to different sub types of `T` based on the name in this field.